### ISSUE TITLE
Suggestion/Idea: non-disableable rules

### ISSUE THREAD

**Issue by chrismbarr** [palantir/tslint#3805]
I would love to have the ability to ignore all rule disables and just report on all violations found regardless of these.  I could see this being a configurable option from the CLI like
```
tslint --no-disable files/**/*.ts
````
to run it this way one-time, or possibly an option per-rule in case there's something important that should never be worked around:
```json
"no-eval": {
  "noDisable": true
}
```

So with either one of those, the following code would still produce a violation regardless of the line that disables it:
```ts
//tslint:disable-next-line
eval("alert('hi')");
```

--------------

I would love to see something like this to help keep my codebase always at a high-level of quality.  Linting is great, but on a team of 30+ developers, any one of them can get lazy and just disable a rule and then no one ever knows about the potential issue again.

**Comment by JoshuaKGoldberg** [comment]
### 💀 _**It's time!**_ 💀

**TSLint is being deprecated** and no longer accepting pull requests for major new changes or features. See #4534. 😱

If you'd like to see this change implemented, you have two choices:
* **Recommended:** Check if this is available in [ESLint](https://eslint.org) + [typescript-eslint](https://typescript-eslint.io) ✅
* _Not Recommended: Fork TSLint locally_ 🤷‍♂️

👋 It was a pleasure open sourcing with you!

_If you believe this message was posted here in error, please comment so we can re-open the issue!_

**Comment by JoshuaKGoldberg** [comment]
🤖 Beep boop! 👉  **TSLint is deprecated** 👈 _(#4534)_  and you should **switch to [typescript-eslint](https://typescript-eslint.io)**! 🤖 

🔒 This issue is being locked to prevent further unnecessary discussions. Thank you! 👋
